sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/utils/dataProcessing.scala

```diff
@@ -0,0 +1,178 @@
+package net.sansa_stack.ml.spark.clustering.utils
+
+import java.io.{File, FilenameFilter}
+
+import com.typesafe.config.Config
+import org.apache.jena.graph.Triple
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+
+import net.sansa_stack.ml.spark.clustering.datatypes.{Categories, CoordinatePOI, Poi}
+import net.sansa_stack.rdf.spark.io.NTripleReader
+
+
+
+/**
+  * load TomTom dataset
+  * @param spark SparkSession
+  * @param conf Configuration
+  */
+class dataProcessing(val spark: SparkSession, val conf: Config) extends Serializable {
+
+  // val dataRDD: RDD[Triple] = NTripleReader.load(spark, conf.getString("sansa.data.input")).persist()
+  val dataRDD: RDD[Triple] = loadNTriple(conf.getString("sansa.data.input"))
+
+  // var poiCoordinates: RDD[(Long, Coordinate)] = this.getPOICoordinates(16.192851, 16.593533, 48.104194, 48.316388).sample(withReplacement = false, fraction = 0.01, seed = 0)
+  var poiCoordinates: RDD[(Long, CoordinatePOI)] = this.getPOICoordinates
+  var poiFlatCategoryId: RDD[(Long, Long)] = this.getPOIFlatCategoryId
+  var poiCategoryId: RDD[(Long, Set[Long])] = this.getCategoryId(poiCoordinates, poiFlatCategoryId).persist()
+  var poiCategoryValueSet: RDD[(Long, Categories)] = this.getCategoryValues  // (category_id, Categories)
+  var poiCategories: RDD[(Long, Categories)] = this.getPOICategories(poiCoordinates, poiFlatCategoryId, poiCategoryValueSet)  // (poi_id, Categories)
+  val poiYelpCategories: RDD[(Long, (Categories, Double))] = this.getYelpCategories(dataRDD).sample(withReplacement = false, fraction = 0.1, seed = 0)
+  var pois: RDD[Poi] = { if (!poiYelpCategories.isEmpty()) {
+    // val poiAllCategories: RDD[(Long, Categories, Double)] = poiCategories.join(poiYelpCategories).map(x => (x._1, (Categories(x._2._1.categories++x._2._2._1.categories), x._2._2._2))
+    val poiAllCategories: RDD[(Long, (Categories, Double))] = poiYelpCategories.join(poiCategories).map(x => (x._1, (Categories(x._2._1._1.categories++x._2._2.categories), x._2._1._2)))
+    poiCoordinates.join(poiAllCategories).map(x => Poi(x._1, x._2._1, x._2._2._1, x._2._2._2)).persist()
+  } else {
+    println("--------pois--------------")
+    poiCoordinates.join(poiCategories).map(x => Poi(x._1, x._2._1, x._2._2, 0.0)).persist()
+  }}
+
+  def loadNTriple(tripleFilePath: String): RDD[Triple] = {
```

Comment:
um, not sure why we have yet another NT reader introduced here. We have our own [io readers](https://github.com/SANSA-Stack/SANSA-RDF/blob/develop/sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/io/package.scala#L182) which could also read a single file or a bunch of them by providing a directory instead of a file name.
Does it check for special files which contain SUCCESS and ends with .crc ? or this is just to check the successfully parsed files (containing triples)?
I think we have to discuss this part more.